src/main/scala/it/pagopa/interop/agreementprocess/api/impl/AgreementApiServiceImpl.scala

```diff
@@ -821,31 +821,22 @@ final case class AgreementApiServiceImpl(
   private def verifyConflictingAgreements(
     agreement: AgreementManagement.Agreement,
     conflictingStates: List[AgreementManagement.AgreementState]
-  )(implicit contexts: Seq[(String, String)]): Future[Unit] = verifyConflictingAgreements(
-    agreement.producerId,
-    agreement.consumerId,
-    agreement.eserviceId,
-    agreement.descriptorId,
-    conflictingStates
-  )
+  )(implicit contexts: Seq[(String, String)]): Future[Unit] =
+    verifyConflictingAgreements(agreement.consumerId, agreement.eserviceId, conflictingStates)
 
   private def verifyConflictingAgreements(
-    producerId: UUID,
     consumerId: UUID,
     eServiceId: UUID,
-    descriptorId: UUID,
     conflictingStates: List[AgreementManagement.AgreementState]
   )(implicit contexts: Seq[(String, String)]): Future[Unit] = {
     for {
       activeAgreement <- agreementManagementService.getAgreements(
-        producerId = Some(producerId.toString),
```

Comment:
filter for producer AND eservice is redundant.
We just need to check that the condition is verified on the tuple (consumer, eservice)